### ISSUE TITLE
[bug] `nemotron_omni_collate_fn` skips packing metadata when image-text processor returns `[1, ΣL]`

### ISSUE THREAD

**Issue by ShiChenAI** [NVIDIA-NeMo/Megatron-Bridge#4234]
### Problem

`nemotron_omni_collate_fn` appears to silently skip sequence-packing metadata generation for some image-text microbatches when `pack_sequences=True` and `micro_batch_size > 1`.

For my data, two image-text examples in the same microbatch are concatenated into a single row:

```text
input_ids.shape == [1, 3544]
```

However, the returned batch does not include:

```text
cu_seqlens
cu_seqlens_unpadded
max_seqlen
```

and `position_ids` is a continuous ramp over the whole concatenated sequence:

```text
0..3543
```

instead of resetting per original sample.

In this case, `pack_sequences=True` produces the same output as `pack_sequences=False`:

```text
pack=False shape=(1, 3544) cu_seqlens=False pos_resets=[] max_pos=3543
pack=True  shape=(1, 3544) cu_seqlens=False pos_resets=[] max_pos=3543
```

The problematic condition seems to be the packing guard inside `nemotron_omni_collate_fn`:

```python
if pack_sequences and batch["input_ids"].shape[0] > 1:
```

This assumes that the pre-packed batch is still in padded `[B, S]` form. However, in the combined image-text processor path, multiple examples may already have been concatenated into `[1, ΣL]` before this guard is reached.

I instrumented the function immediately before the guard and observed:

```text
[debug] pre-guard input_ids.shape=(1, 3544) pack_sequences=True num_examples=2
```

So even though the original microbatch contains two examples, `batch["input_ids"].shape[0]` is already `1`, causing the packing branch to be skipped.

This failure is silent: training continues and loss decreases, but the packed samples appear to be treated as one continuous causal sequence unless another downstream mechanism blocks cross-sample attention. I observed degraded / incoherent fine-tuned model outputs, which led me to inspect the collate output.

### Minimal repro

```shell
Call `nemotron_omni_collate_fn` in the same way `VLMConversationDataset` binds it:


functools.partial(nemotron_omni_collate_fn, processor, pack_sequences=True)


with two image-text conversations in the same microbatch:


from megatron.bridge.data.vlm_datasets.collate import nemotron_omni_collate_fn

# exs = [example_0, example_1]
# Each example is a {"conversation": [...]} dict and contains one image.
# The two examples have token lengths 623 and 2921.

for flag in (False, True):
    b = nemotron_omni_collate_fn(exs, processor, pack_sequences=flag)

    pp = b["position_ids"][0].tolist()
    resets = [i for i in range(1, len(pp)) if pp[i] < pp[i - 1]]

    print(
        "pack=%s shape=%s cu_seqlens=%s pos_resets=%s max_pos=%d"
        % (
            flag,
            tuple(b["input_ids"].shape),
            b.get("cu_seqlens") is not None,
            resets[:5],
            max(pp),
        )
    )


Observed output:


pack=False shape=(1, 3544) cu_seqlens=False pos_resets=[] max_pos=3543
pack=True  shape=(1, 3544) cu_seqlens=False pos_resets=[] max_pos=3543


Additional instrumentation immediately before the packing guard:


[debug] pre-guard input_ids.shape=(1, 3544) pack_sequences=True num_examples=2


Relevant code path:


if pack_sequences and batch["input_ids"].shape[0] > 1:
    ...
    pos_flat = torch.cat([torch.arange(L, ...) for L in lengths], ...)
    ...
    batch["cu_seqlens"] = cu_t
    batch["cu_seqlens_unpadded"] = cu_t.clone()
    batch["max_seqlen"] = max_t


This branch is not reached because `batch["input_ids"].shape[0]` is already `1`.
```

### Expected behavior

If `pack_sequences=True` and multiple original examples are packed into a single `[1, ΣL]` row, the collate function should preserve sample-boundary metadata.

For example, for two samples with lengths 623 and 2921, I would expect:

```text
input_ids.shape:        [1, 3544]
cu_seqlens:             [0, 623, 3544]
cu_seqlens_unpadded:    [0, 623, 3544] or equivalent
max_seqlen:             2921
position_ids:           0..622, 0..2920
```

In other words, `[1, ΣL]` is fine as the packed tensor shape, but it should be accompanied by `cu_seqlens` / `max_seqlen` or equivalent metadata so the packed samples are treated as separate sequences.

A possible fix direction would be to build packing metadata from the known original per-example lengths, regardless of whether the intermediate processor output is `[B, S]` or already `[1, ΣL]`.

For example, the guard could be based on the original sample count or per-example lengths:

```python
if pack_sequences and len(per_sample_lengths) > 1:
    ...
```

rather than:

```python
if pack_sequences and batch["input_ids"].shape[0] > 1:
    ...
```

### Affected area

area:training

### Regression?

Not sure

### Environment

Branch / commit:

```text
NVIDIA-NeMo/Megatron-Bridge
branch: nemotron_3_omni
commit: 648756c
```

Container or environment:

```text
nvcr.io/nvidia/nemo:26.04.00
```

GPU / machine:

```text
single node, 8 GPUs
TP=2 / EP=8 / PP=1 / CP=1
```

Model:

```text
nvidia/Nemotron-3-Nano-Omni-30B-A3B-Reasoning-BF16
```

Relevant file:

```text
src/megatron/bridge/data/vlm_datasets/collate.py
```

Relevant function:

```text
nemotron_omni_collate_fn
```

Python / CUDA / PyTorch:

```text
Using versions provided by nvcr.io/nvidia/nemo:26.04.00
```

### Logs

```shell

```
